### ISSUE TITLE
Any possible way to make this work on Banana Pi Zero?(Unresolved, but probable)

### ISSUE THREAD

**Issue by DeemsDeems** [Vegz78/McAirpos#45]

**Comment by DeemsDeems** [comment]
Sorry, I'm new to Github, wasn't sure how to open a topic on here. I'm trying to do an art project using Makecode Arcade and the only hardware I have access to is Banana Pi Zero instead of Raspberry Pi Zero and I just wanted to know if McAirpos could work with the Banana somehow or is this completely impossibility? Thank you so much in advance for any info.

**Comment by edthedev** [comment]
I've been playing with Libre, and where I had to adjust was I had to use a base image provided by Libre (rather than Raspberry Pi) that worked on Libre. Once the alternate base image was installed with Raspberry Pi Imager, and the system booted up, everything else I have tried has worked the same for me. 

I haven't tried this with McAirpos yet, so your mileage may vary.

But my best guess is it would work fine, if you can can find a Batocera or RecalBox image compatible with BananaPi. 

That said, I'm not quickly finding support from Batocera or RecalBox on BananaPi.

Recalbox FAQ claims support for only Raspberry Pi and Odroid devices:
https://www.recalbox.com/faq/

Batocera system compatibility chart claims support for OrgangePi for some emulators:
https://batocera.org/compatibility.php

I'm inclined to think OrgangePi with Batocera has a good chance of working with McAirpos without paying current Raspberry Pi prices. But again, I haven't tried this yet.

When I build my next arcade machine, I will try some of these options, as I no longer use Raspberry Pi as my primary micro-computer. 

TL;DR: I'm still trying to figure this out myself. OrangePi with Batocera looks the most promising.

**Comment by edthedev** [comment]
Also looks like this could work, but with a disclaimer about sound support:
https://forum.banana-pi.org/t/retropie-on-banana-pi-m1-sd-image-manual-install/11303 

A lot of the alternate Pi hardware providers are coming up with good support with Lakka, so trying out McAirpos on Lakka is on my todo/someday list, myself.

**Comment by Vegz78** [comment]
Hi @DeemsDeems,

Thanks for trying McAirpos! And thank you very much, @edthedev, for replying with your insights!

As I have no other microcontrollers outside RPis, myself, it would be hard to support you in any detail, but great if you could figure out this together and report back here and to the community if it works! Please also feel free to make your own fork of this repository for Banana Pi or other boards, if anything can be reused, or make a pull request if it is possible to include it in McAirpos without breaking any RPi support working today.

All I can add of knowledge regarding your project at this moment is:
-In theory, both MCA games and McAirpos(can be compiled on platform and Python part of `uinput-mapper` should work) should be general ARM Linux compatible, but
-as was the case for some attempts on the Odroid forum, there was another implementation for the frame buffers on some devices, which did not work with MCA games, and
-there were missing dependencies in their Linux distros, and
-different sound issues, as @edthedev already mentions.

So certainly not impossible, given ARM and Linux, but some obstacles on the way is to be expected!

Good luck and hope to hear back here about progress!

Br,
Vegard

**Comment by Vegz78** [comment]
Hi again @DeemsDeems and @edthedev,

Any news or experiences to share regarding running MakeCode Arcade games with or without McAirpos on Libre, OrangePi, BananaPi or inside Lakka or other retro gaming systems?

**Comment by Vegz78** [comment]
Closing this as unresolved due to missing response/presumed lost interest by this issue's OP.

If @DeemsDeems and @edthedev has any news or experiences to share regarding running MakeCode Arcade games with or without McAirpos on Libre, OrangePi, BananaPi or inside Lakka or other retro gaming systems, it would nevertheless be appreciated.
